File: Array & Hashing/1. 两数之和/1. 两数之和.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = {}
        for i, v in enumerate(nums):
            if v in dic:
                return [dic[v], i]
            temp = target - v#補數為key, index為value。後續若有數等於補數時即找到答案
            dic[temp] = i
        return []

File: Array & Hashing/1. 两数之和/1. 两数之和.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> dic;
        vector<int> res;
        for(int i = 0; i < nums.size(); i++){
            if (dic.find(nums[i]) != dic.end()){
                res.push_back(dic[nums[i]]);
                res.push_back(i);
                break;
            }
            int diff = target- nums[i];
            dic[diff] = i;
        }
        return res;
    }
};